nrf_802154_sl/CHANGELOG.rst

```diff
@@ -1,55 +0,0 @@
-.. _nrf_802154_sl_changelog:
```

Comment:
Is it intentional to remove documentation files? I think they are useful here

Reply:
I wanted to bring those files up in subsequent PRs. If you think it is better to do it right away, then it can be done.

Reply:
Let's discuss with tech writers if it is OK to remove these files temporarily. I have no strong opinion

Reply:
@FrancescoSer, what do you think?

Reply:
Hi! I would leave them in place, for now, especially because these are already in RST (unless there is any specific technical reason for not doing so). The remaining doc pages in .dox will be ported also to RST in the near future.

Reply:
I can put those files back. Do you have any suggestions where the documentation should be located in new layout? Should there be a dedicated `doc` directory where RST files for driver/sl/serialization are located?

Reply:
I have added the `CHANGELOG` and `README`. Those still describe the SL though, but both can be refactored later to reflect the new reality.

Reply:
If the SL folder will be contained inside the RD folder, I would structure the doc folders like the ones for the crypto libraries, with a main readme in the root folder, the changelog in the lib folder, and most of the descriptive pages within a doc folder.

By the way, it can stay like it is, for now, as it wouldn't make sense to make this change till the rest of the RD documentation gets converted to RST.

Reply:
I deleted my previous message. After the short catch up with Rafal yesterday I believe that the best course of action would be to remove all .rst files from here (including the readme, changelog and SL related ones).

Reply:
@FrancescoSer , what's the intended documentation structure for `nrf_802154` directory?
I assume it's:

```
nrf_802154/README.rst
nrf_802154/CHANGELOG.rst
nrf_802154/docs/ (documentation pages)
```

Do we want more detailed files like
```
nrf_802154/driver/README.rst
nrf_802154/sl/README.rst
nrf_802154/serialization/README.rst
nrf_802154/driver/CHANGELOG.rst
```
?

Reply:
The current one is:

nrf_802154/README.rst
nrf_802154/ (documentation pages)
nrf_802154/docs/CHANGELOG.rst

Mostly following the nrfmodem structure, so the SL docs are inside the docs folder.

Having a page about the serialization library is actually something I was thinking about, as there isn't any at the moment.
I think it is necessary to have one.

Reply:
Looks good. Thanks for clarificaiton